sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -705,12 +707,15 @@ public String getId() {
             final Mono<TransactionalBatchResponse> responseMono = executor.executeAsync();
 
             return database.getClient().getTracerProvider().
-                traceEnabledBatchResponsePublisher(
-                    responseMono,
-                    context,
-                    this.batchSpanName,
-                    database.getId(),
-                    database.getClient().getServiceEndpoint());
+                traceEnabledBatchResponsePublisher(responseMono,
+                context,
+                this.batchSpanName,
+                this.getId(),
+                database.getId(),
+                database.getClient(),
+                transactionalBatchRequestOptions.getConsistencyLevel(),
```

Comment:
NIT - I personally find the following easier to read

        return database
            .getClient()
            .getTracerProvider()
            .traceEnabledCosmosItemResponsePublisher(
                responseMono,
                context,
                this.createItemSpanName,
                getId(),
                database.getId(),
                database.getClient(),
                ModelBridgeInternal.getConsistencyLevel(options),
                OperationType.Batch,
                ResourceType.Document);